### PR TITLE
Added getInitialNotification method

### PR DESCRIPTION
**Pull request by supermock**
*Issue #928, Feature request*

Good afternoon!

I just implemented the **getInitialNotification()** method, because I need to use it in my work, I do not have time to implement the other features described in this Issue, however I'm marking it here so that you can remove *getInitialNotification* from this **Issue**, and there is no reworking.

Thanks in advance, and I'm waiting: D

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by powerful23** [comment]
fixes #1245 
fixes #928

**Comment by powerful23** [comment]
Hi, thanks for you contribution. I just tested it with Android. Everything works perfect except I can't get the notification data from `getInitialNotification()`. Can you verify that you can receive those data? Thanks!

**Comment by powerful23** [comment]
@supermock Hi, so I tried with the following code:
```js
Amplify.configure(aws_exports);
PushNotification.configure(aws_exports);
window.LOG_LEVEL = 'DEBUG';

PushNotification.onNotification((notification) => {
  console.log('on notification', notification);
});

type Props = {};
class App extends Component<Props> {

  componentDidMount() {
    //  At this point, I already have the application mounted with an instantiated activity. Taking into 
    //  account that in most cases, the developer will want to redirect to another point of the application, 
    //  what time better than that. I also did the test with the 'componentWillMount', which gives a good 
    //  result as well.
    PushNotification.getInitialNotification().then((data) => {
      // Notification data or null if doest have initialized with a notification
      console.log('on initial notification', data);
    });
  }

  render() {
    return (
      <View style={styles.container}>
        <S3Album />
      </View>
    );
  }
}

export default App;
```

I killed the app and then sent a notification from the pinpoint console. I saw the console logging `on notification: {}` but not the `on initial notification`. Is it an expected behavior?

**Comment by supermock** [comment]
Yes it is an expected behavior because the **getInitialNotification()** method is for picking up the notification that opened the application, in the case the notification was touched by the user and at this time the componentDidMount was not executed because the application was not opened.

In the case of iOS it would be the same behavior:
https://facebook.github.io/react-native/docs/pushnotificationios#getinitialnotification

**Comment by supermock** [comment]
@powerful23 Sorry for the delay, but I did not have time to see it, but I managed to fit it now. Problem solved, had not dealt with if the app was in the background, because if it is, by clicking on the notification it was not injected into the Intent. But when the app is dead, the notification opens and injects the data.

I'm waiting for the merge :)

File: packages/pushnotification/android/src/main/java/com/amazonaws/amplify/pushnotification/modules/RNPushNotificationCommon.java [hunk 1]
```diff
@@ -49,4 +50,14 @@ public static JSONObject convertJSONObject(Bundle bundle) throws JSONException {
         }
         return json;
     }
+
+    public static Bundle getNotificationBundleFromIntent(Intent intent) {
+        Bundle bundle = null;
+        if (intent.hasExtra("notification")) {
+            bundle = intent.getBundleExtra("notification");
+        } else if (intent.hasExtra("google.message_id")) {
```
Comment (powerful23) [inline packages/pushnotification/android/src/main/java/com/amazonaws/amplify/pushnotification/modules/RNPushNotificationCommon.java]: why would you need to check `google.message_id`?

**Comment by schavaLogi** [comment]
> @powerful23 Sorry for the delay, but I did not have time to see it, but I managed to fit it now. Problem solved, had not dealt with if the app was in the background, because if it is, by clicking on the notification it was not injected into the Intent. But when the app is dead, the notification opens and injects the data.
> 
> I'm waiting for the merge :)

I verified now if I call this new method in this below scenario we are not getting notification information.
1. App killed
2. Received notification and user tap on notification to launch app
3. Application launched but getInitialNotification is null in Splash container Component
Note: StackNavigator screen has initial route, it has componentDidMount within that we used this method to know the getInitialNotification always return null

**Comment by powerful23** [comment]
@supermock sorry to be late. We added a new event handler `onNotificationOpened` in order to get those tapped notifications: https://aws-amplify.github.io/docs/js/push-notifications#working-with-the-api This should be equivalent to the `getInitialNotification` method. Can you take a look to see if it meets your requirement?

**Comment by supermock** [comment]
@powerful23 As I'm not currently developing the application (working on another project, for now) I can not test, but if it does the same thing that was proposed with getInitialNotification (), it should be fine ...

**Pull request closed without merging**